### ISSUE TITLE
For `toolbar top` component, if a badge is present on an element, it's not read by VoiceOver

### ISSUE THREAD

**Issue by pya35** [Orange-OpenSource/ouds-ios#1565]
### Prerequisites

- [x] I have [searched the backlog](https://github.com/Orange-OpenSource/ouds-ios/issues) for duplicate or closed feature requests

### Your test device

iPhone 17

### Device OS version

26.5

### Is Liquid Glass disabled

- [ ] Yes, Liquid Glass is disabled (or OS version is older than 26)
- [ ] No, Liquid Glass is enabled (or OS version is 26 or newer)

### OUDS iOS library version

2.2.0

### Describe the issue

On toolbar top component, badge (empty or count) is not read by Voice Over

<img width="278" height="54" alt="" src="https://github.com/user-attachments/assets/c8cdfac3-fbe9-4d50-989b-d32394c8eae8" />

### Expected Behavior

Badge shall be read

### Accessibility settings

Voice Over

**Comment by pylapp** [comment]
@pya35 Value for _count badge_ is read by _Voice Over_ (tested with current code base and double-checked with _Accessibility Inspector_)

Tested on iPhone 14 Pro with iOS 27 beta

<img height="500" alt="Image" src="https://github.com/user-attachments/assets/4454512e-90ee-4efe-81df-ec76000a37da" />

**Comment by pylapp** [comment]
## Test Cases

cc @ludovic35 

> [!IMPORTANT]
> Conclusion:
> - Sometimes count badge can be focused after the button, sometimes not, for the same codebase (for the component)
> - Cannot add dedicated value for **system** standard badge
> - Count badge value always accessible
> - But indeed nothing done for the standard badge

> [!TIP]
> As we use two types of standard badges (system and OUDS),  users must add details about the standard badge in the associated item (in hint or label).

### iPhone Xs Max (simu), iOS 18.6

- Debug version with current code base compiled with Xcode 26.5
- OUDS badge is used as Liquid Glass not available
- Standard badge on toolbar top: no value / hint / label associated to the badge on the button, badge not found by _Accessibility Inspector_
- Count badge on toolbar top: badge can be focused with _Accessibility Inspector_ but separated to the button

### iPhone 14 Pro, iOS 27 (beta), with Liquid Glass disabled

- Debug version with current code base compiled with Xcode 26.5
- OUDS badge is used as Liquid Glass disabled (forced)
- Standard badge on toolbar top: no value / hint / label associated to the badge on the button, badge not found by _Accessibility Inspector_, _Voice Over_ does not see the badge
- Count badge on toolbar top: badge can be focused with _Accessibility Inspector_ but separated to the button, _Voice Over_ can focus the badge but separated to the button

> [!NOTE]
> Spotted a bug in that case where no count nor standard badge is displayed, will be fixed, thanks!
> See #1623

### iPhone 14 Pro, iOS 27 (beta), with Liquid Glass enabled

- Debug version with current code base compiled with Xcode 26.5
- System badge is used as Liquid Glass enabled, **because of rendering issues with Liquid Glass and such navigations items**
- Standard badge on toolbar top: no value / hint / label associated to the badge on the button, badge not found by _Accessibility Inspector_, _Voice Over_ does not see the badge, button vocalized
- Count badge on toolbar top: badge is merged in the button which can be focused with _Accessibility Inspector_  _Voice Over_ can focus the button and vocalized the value of the badge

### iPhone 17 Pro, iOS 26.5.1, with Liquid Glass enabled

- AppStore version of the Design System Toolbox (2.2.0)
- System badge is used as Liquid Glass enabled, **because of rendering issues with Liquid Glass and such navigations items**
- Standard badge on toolbar top: no value / hint / label associated to the badge on the button, _Voice Over_ does not see the badge, button vocalized
- Count badge on toolbar top: badge is not merged in the button which can be focused with _Accessibility Inspector_  _Voice Over_ can focus the button and vocalized the value of the badge, and focuses also the badge
